netsim/modules/dhcp.py

```diff
@@ -118,38 +118,93 @@ def check_protocol_support(node: Box, topology: Box) -> bool:
 '''
 
 def build_topology_dhcp_pools(topology: Box) -> None:
-  topology.dhcp.pools = []
+  pools = data.get_empty_box()
+
+  # Phase 1: Build a list of potential pools from VLANs
+  #
+  for vname,vdata in topology.get('vlans',{}).items():
+    if vdata.get('mode') == 'route':                        # Won't deal with routed VLANs
+      continue
+
+    pid = strings.make_id(f'vlan_{vname}')
+    for af in log.AF_LIST:
+      af_pfx = vdata.get(f'prefix.{af}',None)
+      if isinstance(af_pfx,str):
+        pools[pid][af] = af_pfx
+
+      if vdata.get(f'gateway.{af}'):
```

Comment:
Wondering why you're using ``` vdata.get(f'prefix.{af}',None)``` but ```vdata.get(f'gateway.{af}')``` (without None)

Reply:
Good catch. I don't think anything bad happens (because we're using Box not dict), but it's good to be consistent. Fixed.